etc/doc/PROTOCOL.md

```diff
@@ -93,12 +109,39 @@ interface GoalAnswer<Pp> {
   messages: Pp[];
   error?: Pp;
 }
-
 ```
 
 which can be then rendered by the client in way that is desired.
 
-`proof/goals` was first used in the lambdapi-lsp server implementation, and we adapted it to `coq-lsp`.
+The main objects of interest are:
+- `Hyp`: This represents a pair of hypothesis names and type,
+  additionally with a body as obtained with `set` or `pose` tactics
+- `Goal`: Self-explanatory
```

Comment:
Is it the current goal or the list of goals?

Reply:
Added a bit more info, thanks for the review.